Web3Swift/StringScalar/TrimmedPrefixString.swift

```diff
@@ -38,15 +40,10 @@ public final class TrimmedPrefixString: StringScalar {
     public func value() throws -> String {
         let string = try self.string.value()
         let prefix = try self.prefix.value()
-        if string.hasPrefix(prefix) {
-            return String(
-                string.dropFirst(
-                    Int(prefix.count)
-                )
-            )
-        } else {
-            return string
-        }
+        let pattern = "^(\(prefix)){1,}"
+        guard let range = string.range(of: pattern, options: [.regularExpression]) else { return string }
```

Comment:
I think regex is unnecessary overkill in this case. You can simply do something like (pseudocode): 
```
let suffix = string.suffix(string.count % prefix.count)
while string.hasPrefix(prefix): string.dropFirst(prefix.count)
return string + suffix
```

Reply:
bonus points for immutable (`reduce`) version )

Reply:
It looks less understandable than just one regex